### DISCUSSION TITLE
Would FieldResolver overfetch when the field was already populated?

### DISCUSSION THREAD

**Discussion by Tokenyet** [MichalLytek/type-graphql#836] [Q&A] [answered]
Just a curious about how does `FieldResolver` work, here is the use case for the question.

```
  @Authorized([AccountPermission.Business.toString()])
  @Mutation(() => Mobile)
  async mobileCreate(@Ctx() ctx: Context, @Arg('mobile') mobileInput: CreateMobileInput): Promise<Mobile> {
    const doc= await BusinessModel.findById(new ObjectId(ctx.payload.id)).exec();
    const mobileDoc = new MobileModel();

    doc.driver = businessDoc;
    doc.title = mobileInput.title;
    doc.contact = mobileInput.contact;

    try {
      await doc.save();
    } catch(err) {
      throw new InternalServerError();
    }
    
    return doc;
  }
```

Because I created a `Mobile` and pass a real doc on `driver` field, so It's enough for returning the data and no need `MobileResolver` to call `@FieldResolver` on **driver** field. In my little experience, `@FieldResolver` would still be called to do the job, again, right? Just asking If It's normal to do this on *object creation*.

**Comment by MichalLytek** [comment, marked as answer]
Yes, `FieldResolver` overwrites the default `object[property]` field resolver.

If you can have the value populated in some cases, you can just add `if (value) return value` statement in the field resolver.

**Reply by Tokenyet** [reply]
Thanks for the prompt reply 😆.
